### ISSUE TITLE
Should we expose MAX_CONCURRENT_TASKS as a user setting?

### ISSUE THREAD

**Issue by kaste** [SublimeLinter/SublimeLinter#1341]
In the latest iteration `MAX_CONCURRENT_TASKS = multiprocessing.cpu_count()`. In my smoke tests this is the fastest setting you can set. Obviously lint processes are CPU bound. And with that setting you get 99% load. 

Now if a user would set this to 2 our whatever, the user still has 40% - 70% CPU load but for a longer period. Today computers/laptops can handle short CPU spikes, otherwise a 50% load for 10 seconds will still turn the fan on. 

So basically I'm 👎 for the user setting but I don't have to decide this.

(In the smoke tests my computer or Sublime never became unresponsive, yay. It is probably important to document that you must turn off debug or generally extensive console logging bc the console is really super slow and blocks. It is so slow that Windows thinks Sublime has died.)

**Comment by braver** [comment]
I basically never have performance issues on my machine. However, some ideas:

I don’t want a user setting because I can’t see how a user determines the correct value. If our default is too high for someone, I bet we can do a better default.

I like leaving a core for something else. I’ve got 4 virtual cores, so if I start a heavy job (e.g. cmake) I’ll give it 3 cores so I have one left for Freecell. I don’t think non-critical processes should ever claim all the cores. You’re right, usually it’s better to ride the turbo boost for a second. But slow linters shouldn’t clog up the cpu either.
